### ISSUE TITLE
Block provided by BlockAdded event does not match block specification

### ISSUE THREAD

**Issue by mcdee** [CasperLabs/casper-node#1334]
The block specification in the JSON-RPC API shows the era report containing lists of rewards and validator weights.  This shows up as expected if a block is fetched, e.g.:

```
          "era_report": {
            "equivocators": [],
            "inactive_validators": [],
            "rewards": [
              {
                "amount": 1202458594726,
                "validator": "01026ca707c348ed8012ac6a1f28db031fadd6eb67203501a353b867a08c8b9a80"
              },
              {
                "amount": 19956734910,
                "validator": "010427c1d1227c9d2aafe8c06c6e6b276da8dcd8fd170ca848b8e3e8e1038a6dc8"
              },
```

where `rewards` is an array of reward objects.  However, if a block is received on the events stream with the `BlockAdded` event the `rewards` and `next_era_validator_weights` are returned as maps with the key being the validator public key.

This discrepancy makes life difficult to use a standard process for parsing blocks, and appears to serve no purpose.  Could the block presented with the `BlockAdded` event use the same format at that returned by the JSON-API?

**Comment by EdHastingsCasperLabs** [comment]
Thanks; we're looking into it.

**Comment by Fraser999** [comment]
Thanks @mcdee - the discrepancy should be resolved in our 1.2.0 release.

**Comment by Fraser999** [comment]
Fixed in https://github.com/CasperLabs/casper-node/pull/1346.
